include/exadg/incompressible_navier_stokes/spatial_discretization/viscosity_model_base.cpp

```diff
@@ -0,0 +1,63 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <exadg/incompressible_navier_stokes/spatial_discretization/viscosity_model_base.h>
+
+namespace ExaDG
+{
+namespace IncNS
+{
+template<int dim, typename Number>
+ViscosityModelBase<dim, Number>::ViscosityModelBase()
+  : dealii::Subscriptor(),
+    dof_index_velocity(0),
+    quad_index_velocity_linear(0),
+    viscosity_newtonian_limit(0),
+    matrix_free(nullptr)
+{
+}
+
+template<int dim, typename Number>
+void
+ViscosityModelBase<dim, Number>::initialize(
+  dealii::MatrixFree<dim, Number> const &                matrix_free_in,
+  std::shared_ptr<Operators::ViscousKernel<dim, Number>> viscous_kernel_in,
+  unsigned int                                           dof_index_velocity_in,
+  unsigned int                                           quad_index_velocity_linear_in)
+{
+  matrix_free    = &matrix_free_in;
+  viscous_kernel = viscous_kernel_in;
+
+  dof_index_velocity         = dof_index_velocity_in;
+  quad_index_velocity_linear = quad_index_velocity_linear_in;
+
+  // viscosity in the Newtonian limit
+  Operators::ViscousKernelData viscous_kernel_data = viscous_kernel->get_data();
+  viscosity_newtonian_limit                        = viscous_kernel_data.viscosity;
```

Comment:
The variable `viscosity_newtonian_limit` should in my opinion not exist in `ViscosityModelBase`, because it is redundant code/information.